hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java

```diff
@@ -601,43 +602,86 @@ protected Session session(HugeType tableType) {
+                LOG.debug("The snapshot data path: {}", snapshotPath);
+                RocksDBSessions sessions = entry.getValue();
+                sessions.createSnapshot(snapshotPath.toString());
+
+                uniqueParents.add(snapshotPath.getParent().toString());
             }
+            LOG.info("The store '{}' save snapshot successfully", this.store);
+            return uniqueParents;
         } finally {
-            writeLock.unlock();
+            readLock.unlock();
         }
     }
 
     @Override
-    public void readSnapshot(String parentPath) {
-        Lock writeLock = this.storeLock.writeLock();
-        writeLock.lock();
+    public Set<String> resumeSnapshot(String snapshotPrefix) {
+        Lock readLock = this.storeLock.readLock();
+        readLock.lock();
         try {
+            Set<String> uniqueParents = new HashSet<>();
             if (!this.opened()) {
-                return;
+                return uniqueParents;
             }
 
-            File[] snapshotFiles = new File(parentPath).listFiles();
-            E.checkNotNull(snapshotFiles, "snapshot files");
-            List<Pair<File, File>> fileRenamePairs = new ArrayList<>();
-            for (File snapshotFile : snapshotFiles) {
-                Session session = this.findMatchedSession(snapshotFile);
-                File dataFile = new File(session.dataPath());
-                fileRenamePairs.add(Pair.of(snapshotFile, dataFile));
+            Map<Path, Path> fileNameMaps = new HashMap<>();
+            for (Map.Entry<String, RocksDBSessions> entry : this.dbs.entrySet()) {
+                // Like: parent_path/rocksdb-data/m
+                //       parent_path/rocksdb-vertex/g
+                Path originDataPath = Paths.get(entry.getKey()).toAbsolutePath();
+                Path parentParentPath = originDataPath.getParent().getParent();
+                // Like: rocksdb-data/m
+                //       rocksdb-vertex/g
+                Path pureDataPath = parentParentPath.relativize(originDataPath);
+                // Like: parent_path/snapshot_rocksdb-data/m
+                //       parent_path/snapshot_rocksdb-vertex/g
+                Path snapshotPath = parentParentPath.resolve(snapshotPrefix +
+                                                             "_" + pureDataPath);
+                E.checkState(snapshotPath.toFile().exists(),
+                             "The snapshot path '%s' doesn't exist",
+                             snapshotPath);
+                LOG.debug("The origin data path: {}", originDataPath);
+                LOG.debug("The snapshot data path: {}", snapshotPath);
+
+                RocksDBSessions sessions = entry.getValue();
+                RocksDB rocksdb = sessions.createSnapshotRocksDB(
+                                  snapshotPath.toString());
+                Path snapshotLinkPath = Paths.get(originDataPath + "_link");
+                createCheckpoint(rocksdb, snapshotLinkPath.toString());
```

Comment:
why createCheckpoint in resumeSnapshot()

Reply:
when under raft mode, need reserve the origin snapshot dir, so need create hard link from snapshot